### PR TITLE
Rename noop to nop, add nop exporter, rename TraceDataExporter to TraceExporter.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
This makes TraceExporter consistent with TraceReceiver. Maybe we should change TraceDataProcessor to TraceProcessor?

File: exporter/exportertest/sink_exporter.go [hunk 1]
```diff
@@ -12,55 +12,60 @@
...
 	mu      sync.Mutex
 	metrics []data.MetricsData
 }
 
-var _ processor.MetricsDataProcessor = (*ConcurrentMetricsDataSink)(nil)
+var _ exporter.MetricsExporter = (*SinkMetricsExporter)(nil)
 
 // ProcessMetricsData stores traces for tests.
-func (cms *ConcurrentMetricsDataSink) ProcessMetricsData(ctx context.Context, md data.MetricsData) error {
+func (cms *SinkMetricsExporter) ProcessMetricsData(ctx context.Context, md data.MetricsData) error {
```
Comment (pjanotti) [inline exporter/exportertest/sink_exporter.go]: nit: same thing cms doesn't match the type anymore

File: exporter/exportertest/sink_exporter.go [hunk 2]
```diff
@@ -12,55 +12,60 @@
...
 }
 
-var _ processor.TraceDataProcessor = (*ConcurrentTraceDataSink)(nil)
+var _ exporter.TraceExporter = (*SinkTraceExporter)(nil)
 
 // ProcessTraceData stores traces for tests.
-func (css *ConcurrentTraceDataSink) ProcessTraceData(ctx context.Context, td data.TraceData) error {
-	css.mu.Lock()
-	defer css.mu.Unlock()
+func (cts *SinkTraceExporter) ProcessTraceData(ctx context.Context, td data.TraceData) error {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply exporter/exportertest/sink_exporter.go]: done.

File: exporter/exportertest/sink_exporter.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/exportertest/sink_exporter.go]: done.

**Pull request merged**